DOC-1742 Cloning auth service and gateway service files (#410)
corda/corda-docs: content/en/docs/corda-enterprise/4.7/node/auth-service.md

---
menu:
  corda-enterprise-4-7:
    parent: corda-enterprise-4-7-corda-nodes
tags:
- authentication
- auth
weight: 150
aliases: "/docs/cenm/1.5/auth-service.html"

title: Auth service
---

> # Auth service

Reviewer: ```suggestion
# Auth Service
```

> The Auth service is the user authentication and authorization service for managing Corda Nodes and networks (CENM). It stores and controls secure user-access to network services, such as:

Reviewer: ```suggestion
The Auth Service is the user authentication and authorization service for managing Corda Nodes and networks (CENM). It stores and controls secure user-access to network services, such as:
```

> The Auth Service is the user authentication and authorization service for managing Corda Nodes and networks (CENM). It stores and controls secure user-access to network services, such as:

Reviewer 2: ```suggestion
The Auth Service is the user authentication and authorisation service for managing Corda Nodes and networks (CENM). It stores and controls secure user-access to network services, such as:
```

> * Nodes

Reviewer 2: ```suggestion
* Nodes.
```
Do we use full stops in unordered lists? I think that's what the style guide says (though I do recall a rather heated debate about it shortly after I joined).

> * Identity manager

Reviewer: ```suggestion
* Identity Manager
```

> * Identity manager

Reviewer 3: Shouldn't be all CENM services grouped together - just to distinguish it from Nodes?

> * Zone service

Reviewer: ```suggestion
* Zone Service
```

> * Signing service

Reviewer: ```suggestion
* Signing Service
```

> * Network map (and associated network configurations and node info)

Reviewer: ```suggestion
* Network Map (and associated network configurations and node info)
```

> Whenever you use the [CENM User admin tool](../../../cenm/1.5/user-admin) to create new users, groups or roles, the Auth service is updated to authenticate those users and their permissions. When using the remote management tools such as the [CENM Command Line Interface](../../../cenm/1.5/cenm-cli-tool) or the web GUIs hosted on the Gateway Service, the auth service verifies your identity and security clearance as needed.

Reviewer: ```suggestion
Whenever you use the [CENM User admin tool](../../../cenm/1.5/user-admin) to create new users, groups or roles, the Auth Service is updated to authenticate those users and their permissions. When using the remote management tools such as the [CENM Command Line Interface](../../../cenm/1.5/cenm-cli-tool) or the web GUIs hosted on the Gateway Service, the Auth Service verifies your identity and security clearance as needed.
```

> You do not need to interact directly with the Auth Service once it has been installed and configured. To protect the integrity of this secure service, there is no direct API contact with the Auth Service: all front-end communications go via the Gateway service.

Reviewer: ```suggestion
You do not need to interact directly with the Auth Service once it has been installed and configured. To protect the integrity of this secure service, there is no direct API contact with the Auth Service: all front-end communications go via the Gateway Service.
```

> Auth service can also be configured to use [Azure AD SSO](azure-ad-sso).

Reviewer: ```suggestion
Auth Service can also be configured to use [Azure AD SSO](azure-ad-sso).
```

To do this:

1. Create a SSL certificate in a `.jks` file using the [CENM PKI tool](../../../cenm/1.5/PKI-tool).

> 2. Generate a `jwt` signing key (RSA keypair) in a jks file with the following command line command:

Reviewer: ```suggestion
2. Generate a JWT signing key (RSA keypair) in a `.jks` file with the following command-line command:
```

In the sample below, you can see the initial configuration process:

> 1. [Database configuration](../../../cenm/1.5/database-set-up). Add the name, address and login credentials for the SQL database that supports the Auth service.

Reviewer: ```suggestion
1. [Database configuration](../../../cenm/1.5/database-set-up). Add the name, address and login credentials for the SQL database that supports the Auth Service.
```

{{<note>}}
If multiple CENM instances are connected to the same database, setting `lockResolutionStrategy` to `SingleInstance` can cause startup problems and/or database corruption. For more information, see the [database configuration options](../../../cenm/1.5/config-database.md).
{{</note>}}

> 2. JSON Web Key configuration. Set the username, password, and location of the RSA keypair store for signing. The location must be the absolute path.

Reviewer: ```suggestion
2. JSON Web Key configuration. Set the user name, password, and location of the RSA keypair store for signing. The location must be the absolute path.
```

> 2. JSON Web Key configuration. Set the username, password, and location of the RSA keypair store for signing. The location must be the absolute path.

Reviewer: ```suggestion
2. JSON Web Key configuration. Set the user name, password, and location of the RSA keypair store for signing. The location must be the absolute path.
```

* `[--reset-user-admin]` Flag to enable forcing the user to be admin.

* `[--keep-running]` Enables the application to keep running after any of the **initializer** arguments have been supplied.

## Setting up applications

{{< note >}}

> Auth service needs to be set up with baseline permission data for each application.

Reviewer: ```suggestion
Auth Service needs to be set up with baseline permission data for each application.
```
